### PR TITLE
Documentation - Update FAQ support

### PR DESCRIPTION
**Pull request by ravital**
https://github.com/nucypher/nucypher/blob/ravital-faq/docs/source/support/faq.rst

### CODE REVIEW

**Comment by derekpierre** [comment]
Thanks for putting this together @ravital !

I was wondering about relabelling the sections for the FAQ - currently `Network-Related Questions` and `Setup-Related Questions`. The reason being that "Setup Related Questions" could apply to both developers/users and stakers, and `Network Related Questions` as a section would presumably have non-setup related questions that may apply to both stakers and developers/users of the network, eg. developers installing `nucypher` to use the API.

We have previously had a difficult time organizing our docs based on the relevant audience and it caused confusion to readers. Eventually, we settled on using a separation of "Running a node" vs "Application Development" or simply "supply" vs "demand" sides of our market for our guides which has helped with clarity. I'm not beholden to those, but I think they have worked out very well for our guides and is something to consider.

For instance, the section on `Setup-Related Question` could become `Running a Node` (or something similar) since it contains the questions from the initial FAQ which was the intended audience when it was written.

`Network Related Questions` could become `Application Development` (or something similar) with the following questions removed:
* Why have the Staker and Ursula been split? -> not applicable to app dev, and already answered for Stakers.
* What kind of token is NU? -> Perhaps it can be changed to `Does Alice or Bob need NU to use the network?`...

Assuming we have a sufficient number of general questions, perhaps we can even add an additional "General" section for common questions that apply to both groups (stakers and developers/users) like:
* How long is a period?
* ...?

Just a thought, what do you (and others) think?

**Comment by KPrasch** [comment]
Nice content additions here! :+1:  - I think it's important to have likable questions, so some follow-up formatting is needed.

**Comment by ravital** [comment]
Sure @derekpierre that makes sense to have it as "Running a Node" for Setup. I'm not sure about "Application Development" as a section title though. Maybe a different title? There's certainly questions there that I would ask (and think are fundamental to understanding the network) if I were reading through our documentation/learning about our network. I think "Application Development" sounds very specialized and a bit misleading. 

I'm sorry I couldn't figure out the formatting nicely though :(

File: docs/source/support/faq.rst [hunk 1]
```diff
@@ -2,29 +2,123 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Ursulas are “semi-trusted” in the sense that Alice must trust Ursula to revoke a policy once it expires or if instructed to do so (by Alice). We also must trust Ursula to be responsive and perform the re-encryption correctly so that Bob can access the data. 
+
+Q: Who pays Ursula? How is it done?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently, Alice pays Ursula (since Ursula is doing work for Alice). Alice pays an ETH deposit into the PolicyManager contract when granting.
+(However, we do imagine Bob paying in some models!)
+
```
Comment (arjunhassard) [inline docs/source/support/faq.rst]:
```suggestion
Q: How much does Alice pay?
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Alice provides an upfront deposit that covers the entirety of the sharing policy's duration, for all the Ursulas required to service the policy. Currently, Ursulas may choose a discretionary payment rate per period, which Alices discover on an Ursula-by-Ursula basis by querying public network contracts. In the early stages of the network, both Alice and Ursula must settle on a rate which falls within a fixed, universal range, enforced by the PolicyManager contract.

```

File: docs/source/support/faq.rst [hunk 2]
```diff
@@ -2,29 +2,123 @@
...
---------------------------------------------------------------------
+Network-Related Questions
+-------------------------
 
-Worker nodes need to run ``nucypher`` and a local ethereum node. In total, you will
-require at least 4GB for RAM. Nodes also need 24/7 uptime and a static, public IPv4 address.
+*These are questions related to how the NuCypher network works.*
 
-For ``nucypher`` specific requirements, see `System Requirements and Dependencies <https://docs.nucypher.com/en/latest/guides/installation_guide.html#system-requirements-and-dependencies/>`_.
+We assume familiarity with the network characters "**Alice**," "**Bob**," "**Ursula**," and "**Enrico**."
```
[the remark this answers is not in the document]
Comment (arjunhassard) [reply docs/source/support/faq.rst]: The FAQ is the first place a lot of visitors will go to, so wished to soften the warning about prerequisite knowledge. In any case, the characters are briefly explained below this.

**Comment by arjunhassard** [comment]
@derekpierre Let's add the following type of question to the user section (from https://github.com/nucypher/nucypher/issues/829): 

Why can't I just 're-encrypt' sym keys in the client?
Why is useful for entities other than the data owner to encrypt on their behalf?
Why can't I just use a smart contract to enforce access control?
Why can't my users re-use their ETH wallet pk/sk when creating a sharing policy?
How do I integrate NuCypher if my application is not written in Python?

**Comment by derekpierre** [comment]
Fixes #1792

**Comment by KPrasch** [comment]
Remaining RFCs can be applied to #1806

**Pull request merged**